### ISSUE TITLE
Errors occur with LaTeX-based reports

### ISSUE THREAD

**Issue by beatngu13** [beatngu13/pdf-zoom-wizard#17]
As already mentioned by a few users (e.g. [here](https://github.com/beatngu13/pdf-zoom-wizard/issues/16#issuecomment-450530059) and [here](https://github.com/beatngu13/pdf-zoom-wizard/issues/15#issue-338987405)), errors occur with LaTeX-based reports.

After some investigation, I found out that the following exception is thrown in [`Wizard#modifyBookmarks(Bookmarks)`](https://github.com/beatngu13/pdf-zoom-wizard/blob/v0.1.1/src/main/java/com/github/beatngu13/pdfzoomwizard/core/Wizard.java#L260) when `Bookmark#getTitle()` is being invoked:

```
java.lang.ClassCastException: org.pdfclown.objects.PdfReference cannot be cast to org.pdfclown.objects.PdfTextString
```

A hotfix might only remove the usage of `Bookmark#getTitle()`, which would be a workaround until PDF Clown is abandoned.

**Comment by beatngu13** [comment]
@reejbee @dogloveroc I have created a beta release with a workaround for this issue. Would you mind [downloading this version](https://github.com/beatngu13/pdf-zoom-wizard/releases/tag/v0.1.2-beta.1) to see if it works for you? If it does, I can release v0.1.2, otherwise I have to try something else. Thanks in advance!

**Comment by dogloveroc** [comment]
Hi Daniel,
Thanks for working on this. I'm happy to help with testing.
I downloaded the .jar file to test it. After double-clicking on it I received a pop-up window with the following error message:
Java Virtual Machine LauncherCould not find the main class: com.github.beatngu13.pdfzoomwizard.Launcher. Program will exit.
Thanks,Ilana


    On ‎Saturday‎, ‎January‎ ‎5‎, ‎2019‎ ‎11‎:‎47‎:‎28‎ ‎AM‎ ‎PST, Daniel Kraus <notifications@github.com> wrote:  
 
 
@reejbee @dogloveroc I have created a beta release with a workaround for this issue. Would you mind downloading this version to see if it works for you? If it does, I can release v0.1.2, otherwise I have to try something else. Thanks in advance!

**Comment by beatngu13** [comment]
> Hi Daniel, Thanks for working on this. I'm happy to help with testing. I downloaded the .jar file to test it. After double-clicking on it I received a pop-up window with the following error message: Java Virtual Machine LauncherCould not find the main class: com.github.beatngu13.pdfzoomwizard.Launcher. Program will exit. Thanks,Ilana

Hi @dogloveroc, thank you for the fast feedback. This may happen if you use a Java version prior to 1.8 or OpenJDK. Could you please try installing [this Java runtime](https://java.com/en/download/) and then launch the Wizard with it?

**Comment by dogloveroc** [comment]
Yes, you were right. The earlier Java has been removed and the PDF Zoom Wizard now appears. The inherit zoom still doesn't seem to get applied when PDF Producer is ESP Ghostscript 815.01. Sorry.

    On ‎Saturday‎, ‎January‎ ‎5‎, ‎2019‎ ‎03‎:‎20‎:‎36‎ ‎PM‎ ‎PST, Daniel Kraus <notifications@github.com> wrote:  
 
 

Hi Daniel, Thanks for working on this. I'm happy to help with testing. I downloaded the .jar file to test it. After double-clicking on it I received a pop-up window with the following error message: Java Virtual Machine LauncherCould not find the main class: com.github.beatngu13.pdfzoomwizard.Launcher. Program will exit. Thanks,Ilana


Hi @dogloveroc, thank you for the fast feedback. This may happen if you use a Java version prior to 1.8 or OpenJDK. Could you please try installing this Java runtime and then launch the Wizard with it?

**Comment by beatngu13** [comment]
> Yes, you were right. The earlier Java has been removed and the PDF Zoom Wizard now appears. The inherit zoom still doesn't seem to get applied when PDF Producer is ESP Ghostscript 815.01. Sorry.

Too bad, might be a different issue, but thanks for your help so far. Would you mind attaching the PDF such that I can reproduce the error?

**Comment by dogloveroc** [comment]
Yes it may be another issue. The utility creating the PDF has been updated in Linux to apply inherit zoom so going forward the property will be there. 

I won’t be able to share the PDF for confidentiality reasons.

**Comment by davidcayll** [comment]
Hi beatngui13,
I am working with a manual with bookmarks.  It originally operated from 17 separate chapter pdf files.  I am combining them into one document for use on my ipad.  I did take one of the chapter files and use **pdfzoomwizard-0.1.2-beta.1.jar** successfully to change all Zoom Levels to Inherit Zoom.  The single chapter pdf is 1MB in size and the total manual is 28MB in size.  Maybe size is the problem?

The maximum file size here is 10MB and I could not attach the whole 28 MB file.  I deleted chapters for a file size of 9 MB which I checked still fails.  I have attached this file for your study.  

Maybe error handling messages would be handy to include in the final application?  This application will be really handy to do the dirty job of correcting all the zoom references.

Thanks,
David
[1996 Manual_A.pdf](https://github.com/beatngu13/pdf-zoom-wizard/files/2828509/1996.Manual_A.pdf)
